src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -279,7 +352,22 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
     if (!table_id)
     {
         if (exception)
-            exception->emplace(ErrorCodes::UNKNOWN_TABLE, "Cannot find table: StorageID is empty");
+        {
+            String exception_message;
+            if (!db_and_table.first)
+            {
+                DatabaseNameHints hints(*this);
+                auto similarDb = hints.findMostSimilarWord(table_id.getAllRegisteredNames(), table_id.getDatabaseName());
+                exception_message = fmt::format("Database {} doesn't exist. Maybe you mean: {}", table_id.getDatabaseName(), similarDb);
+            }
+            else
+            {
+                TableNameHints hints(*this, getContext(), table_id.getDatabaseName());
+                auto similarTb = hints.findMostSimilarWord(table_id.getAllRegisteredNames(), table_id.getTableName());
+                exception_message = fmt::format("Table {} doesn't exist. Maybe you mean: {}", table_id.getNameForLogs(), similarTb);
+            }
+            exception->emplace(exception_message, ErrorCodes::UNKNOWN_TABLE);
+        }
         return {};
     }
```

Comment:
Actually this code should be executed later because if you have an empty `table_id` you cannot provide any hints.